### ISSUE TITLE
No Attribute "_get_distribution_strategy"

### ISSUE THREAD

**Issue by ThomasWarnasch** [tensorflow/tensorboard#3125]
### TL:DR-ERROR

<details>
<summary>Error Log</summary>
File "C:\Users\Thomas\Desktop\Programs\Python36\lib\site-packages\tensorflow_core\python\keras\callbacks.py", line 1532, in set_model
    self.log_dir, self.model._get_distribution_strategy())  # pylint: disable=protected-access
AttributeError: 'Sequential' object has no attribute '_get_distribution_strategy'
</details>

### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
INFO: diagnose_tensorboard.py version d515ab103e2b1cfcea2b096187741a0eeb8822ef

--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=6, micro=5, releaselevel='final', serial=0)
INFO: os.name: nt
INFO: os.uname(): N/A
INFO: sys.getwindowsversion(): sys.getwindowsversion(major=10, minor=0, build=18362, platform=2, service_pack='')

--- check: package_management
INFO: has conda-meta: False
INFO: $VIRTUAL_ENV: None

--- check: installed_packages
Traceback (most recent call last):
  File "C:\Users\Thomas\Downloads\diagnose_tensorboard.py", line 470, in main
    suggestions.extend(check())
  File "C:\Users\Thomas\Downloads\diagnose_tensorboard.py", line 180, in installed_packages
    freeze = pip(["freeze", "--all"]).decode("utf-8").splitlines()
  File "C:\Users\Thomas\Downloads\diagnose_tensorboard.py", line 106, in pip
    return subprocess.check_output(command)
  File "C:\Users\Thomas\Desktop\Programs\Python36\lib\subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "C:\Users\Thomas\Desktop\Programs\Python36\lib\subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['C:\\Users\\Thomas\\Desktop\\Programs\\Python36\\pythonw.exe', '-m', 'pip', '--disable-pip-version-check', 'freeze', '--all']' returned non-zero exit status 120.

--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.2.0a20200106'

--- check: tensorflow_python_version
INFO: tensorflow.__version__: '2.1.0'
INFO: tensorflow.__git_version__: 'v2.1.0-rc2-17-ge5bf8de410'

--- check: tensorboard_binary_path
INFO: which tensorboard: b'C:\\Users\\Thomas\\Desktop\\Programs\\Python36\\Scripts\\tensorboard.exe\r\n'

--- check: addrinfos
socket.has_ipv6 = True
socket.AF_UNSPEC = <AddressFamily.AF_UNSPEC: 0>
socket.SOCK_STREAM = <SocketKind.SOCK_STREAM: 1>
socket.AI_ADDRCONFIG = <AddressInfo.AI_ADDRCONFIG: 1024>
socket.AI_PASSIVE = <AddressInfo.AI_PASSIVE: 1>
Loopback flags: <AddressInfo.AI_ADDRCONFIG: 1024>
Loopback infos: [(<AddressFamily.AF_INET6: 23>, <SocketKind.SOCK_STREAM: 1>, 0, '', ('::1', 0, 0, 0)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 0, '', ('127.0.0.1', 0))]
Wildcard flags: <AddressInfo.AI_PASSIVE: 1>
Wildcard infos: [(<AddressFamily.AF_INET6: 23>, <SocketKind.SOCK_STREAM: 1>, 0, '', ('::', 0, 0, 0)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 0, '', ('0.0.0.0', 0))]

--- check: readable_fqdn
INFO: socket.getfqdn(): 'LAPTOP-0MQ7N9O5'

--- check: stat_tensorboardinfo
INFO: directory: C:\Users\THOMAS~1\AppData\Local\Temp\.tensorboard-info
INFO: .tensorboard-info directory does not exist

--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['C:\\Users\\Thomas\\Desktop\\Programs\\Python36\\lib\\site-packages']; bad_roots (0): []

--- check: full_pip_freeze
Traceback (most recent call last):
  File "C:\Users\Thomas\Downloads\diagnose_tensorboard.py", line 470, in main
    suggestions.extend(check())
  File "C:\Users\Thomas\Downloads\diagnose_tensorboard.py", line 78, in wrapper
    result = fn()
  File "C:\Users\Thomas\Downloads\diagnose_tensorboard.py", line 437, in full_pip_freeze
    "pip freeze --all:\n%s", pip(["freeze", "--all"]).decode("utf-8")
  File "C:\Users\Thomas\Downloads\diagnose_tensorboard.py", line 106, in pip
    return subprocess.check_output(command)
  File "C:\Users\Thomas\Desktop\Programs\Python36\lib\subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "C:\Users\Thomas\Desktop\Programs\Python36\lib\subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['C:\\Users\\Thomas\\Desktop\\Programs\\Python36\\pythonw.exe', '-m', 'pip', '--disable-pip-version-check', 'freeze', '--all']' returned non-zero exit status 120.
``````

</details>

### Approximate Code

<details>
<summary>Code</summary>

from keras.callbacks import TensorBoard

model,Name = generate_model(L,KS1,KL1,KS2,KL2,0,0,0,0,inputres,outputres)
tensorboard = TensorBoard(log_dir='logs\\{}'.format(Name))
#tensorboard = tf.keras.callbacks.TensorBoard(log_dir='logs', histogram_freq=0,write_graph=True, write_images=False)
                            model.fit(traindatain,traindataout,epochs=epochs_num,batch_size=256,validation_data=(testdatain,testdataout),callbacks=[tensorboard])
model.save(Name+'.model')

</details>

I've recently switched from TF 1.12 to TF 2.0.0 (on a completely new machine) and can't get all the libraries to work like they used to.

Thank you for your time helping me.

**Comment by rmothukuru** [comment]
@ThomasWarnasch,
Since you have switched from 1.12 to 2.0, you can use the utility, `tf_upgrade_v2` to automatically convert your code to 2.0, without (almost) any error. Please find [this Tutorial](tf_upgrade_v2) for more details. Thanks!

**Comment by ThomasWarnasch** [comment]
I used the upgrade utility and exactly like my manually updated code - I still get the same error.

Additionally, removing the callback in either version of my ported code allows the model to train normally.

**Comment by rmothukuru** [comment]
@ThomasWarnasch,
Can you please share the code for `generate_model` and other code as well, if any, so that we can reproduce the issue at our end. Also, can you please confirm if you are using Distribution Strategy? Thanks!

**Comment by ThomasWarnasch** [comment]
### Here's a segment of code that reproduces the error I'm experiencing.

<details>
<summary>Code</summary>


    import numpy as np
    import time
    import tensorflow as tf
    import keras
    from keras.models import Sequential
    from keras.layers import Flatten,MaxPooling1D,GlobalAveragePooling1D,Conv2D,Dropout,Dense
    from keras.callbacks import ModelCheckpoint
    from keras.utils import np_utils
    from keras.callbacks import TensorBoard
    from keras import metrics,optimizers
    import pickle



    epochs_num = 2#12000,6000,3000

    inputres = 24
    outputres= 64



    Name = "Fallback"


    #tensorboard = tf.keras.callbacks.TensorBoard(log_dir='logs', histogram_freq=0, write_graph=True, write_images=False)
    #LOG_DIR = "LOGS/"+str(time.time())
    tensorboard = TensorBoard(log_dir='logs\\{}'.format(Name))
    percent_ignore = 0.28#Ignore amount for reducing memory usage.
    len_trainingdata = 400000

    def create_training_data():
        dbifile = open('trainingdatainput.pickle', 'rb')
        dbofile = open('trainingdataoutput.pickle', 'rb')
        counter = 0
        traindatain=[]
        traindataout=[]
        testdatain=[]
        testdataout=[]
        try:
            for memory in range(int(len_trainingdata*(1-percent_ignore))):
                counter+=1
                if counter%10==0:
                    db = pickle.load(dbifile)
                    db2 = pickle.load(dbofile)
                    if db.shape == (inputres,inputres,3) and db2.shape == (outputres,outputres,3):
                        testdatain.append(db)
                       testdataout.append(db2)
                    else:
                        print("CRITICAL ERROR")
                else:
                    db = pickle.load(dbifile)
                    db2 = pickle.load(dbofile)
                    if db.shape == (inputres,inputres,3) and db2.shape == (outputres,outputres,3):
                        traindatain.append(db)
                        traindataout.append(db2)
                    else:
                        print("Critical Error")
                #print(db)
        except:
            print("error or done")
        
            print(len(traindatain),len(traindataout),len(traindatain[0]),len(traindataout[0]),len(traindatain[0][0][0]),len(traindataout[0][0][0]))
            print(len(testdatain),len(testdataout),len(testdatain[0]),len(testdataout[0]))
            #print(testdata[0])
        return [testdatain],[testdataout],[traindatain],[traindataout]

    def generate_model(inputres,outputres):#,inputres,outputres
        model = None
        Name = "NAMETEST"
        print(Name)
    
        model = Sequential()
        model.add(Conv2D(kernel_size=(3,3),filters=(12),padding="SAME",input_shape = (inputres,inputres,3), activation = "relu"))#16#12 #Same/Valid
        model.add(Flatten())
        model.add(Dense(outputres*outputres*3, activation = "relu"))#,input_shape = traindatain[1,1:].shape
        opti = optimizers.adadelta(lr=.1,decay=1e-6)#Changes learning rate.
        model.compile(loss='mean_squared_error', optimizer=opti,metrics=['mae', 'acc'])
        print(model.summary())
    
        return model,Name

    def train_network(tensorboard,testdatain,testdataout,traindatain,traindataout,epochs_num,inputres,outputres):#input==output as it's a compression network ><
        #Reformats data for network input
        train_length = len(traindataout[0])
        test_length = len(testdataout[0])
        traindataout = np.asarray(traindataout[0])
        traindataout = traindataout.reshape(train_length ,outputres*outputres*3)
        testdataout = np.asarray(testdataout[0])
        testdataout = testdataout.reshape(test_length ,outputres*outputres*3)

        model,Name = generate_model(inputres,outputres)
        model.fit(traindatain,traindataout,epochs=epochs_num,batch_size=256,validation_data=(testdatain,testdataout),callbacks=[tensorboard])
        model.save(Name+'.model')
    
    
    testdatain,testdataout,traindatain,traindataout = create_training_data()
    print("done processing input data")
    train_network(tensorboard,testdatain,testdataout,traindatain,traindataout,epochs_num,inputres,outputres)


</details>

Also can you please clarify what you mean by "Distribution Strategy"? I'm not clear of what this means in this specific context.
